File: TD/TD3/Code/7.cpp
/* Dessiner un carré de côté N à l'écran */
#include <iostream>

void dessin(int a, char c)
{
  for (int i = 1; i <= a; ++i) {
    for (int j = 1; j <= a; ++j) {
      if (i == 1 || i == a || j == 1 || j == a) {
        std::cout << c << std::flush;
      } else {
        std::cout << " " << std::flush;;
      }
    }
    std::cout << std::endl;
  }
}

int main ()
{
  int a;
  char c;
  
  std::cout << "Donne moi une valeur : ";
  std::cin >> a;
  std::cout << "Donne moi un caractere : ";
  std::cin >> c;
  
  dessin(a, c);
  
  return 0;
}

File: TD/TD3/Code/Python/7.py
#!/usr/bin/python
# Dessiner un carré de côté N à l'écran


def dessine_carre(nmbr, car):
    for i in range(1, nmbr + 1):
        for j in range(1, nmbr + 1):
            if i == 1 or j == 1 or i == nmbr or j == nmbr:
                print(car, end='', sep='')
            else:
                print(' ', end='', sep='')
        print('\n', end='')


print('Donne moi une valeur : ', end='')
val = int(input())
print('Donne moi un caractère : ', end='')
car = input()

dessine_carre(val, car)
